### DISCUSSION TITLE
Triggering a page reload from a server

### DISCUSSION THREAD

**Discussion by festeh** [sveltejs/kit#11733] [Q&A] [answered]
Hi!

I do something weird - a desktop app in Svelte with [tauri](https://tauri.app/). So the client and server live on a same machine and have a single user. 

My app has a POST endpoint and when it receives a request, it changes the app state on the server. How to notify client of this
change and force it to automatically reload the page?

ChatGPT suggests to poll server from a client every second, if state has changed, do `invalidate_all()`. Is there a less hacky way?

Thanks for your help!

**Comment by david-plugge** [comment, marked as answer]
Well, since you are using tauri you should use a builtin mechanism for this kind of stuff. In this case you can use events: https://tauri.app/v1/guides/features/events

Emit an event from the server and listen for that on the client. In the callback you should call invalidateAll, which is the correct way to trigger a rerun of all load functions.

**Reply by david-plugge** [reply]
Actually... If you are doing a post request from the tauri client to its server, can't you simply call invalidateAll after receiving the response on the client? I'd suggest trying that first as it's even easier to implement and understand.

**Reply by festeh** [reply]
Thanks for your answer! I honestly expected a pure Sveltekit solution, since that is not a Tauri forum... Since I have not stated it clearly, it is fair to mark your answer as a response.

Should say Tauri docs are not very helpful, as there is only an example of listening for a "click" event from the JS, in other examples Rust backend listens for JS.

Regarding your second message, a POST request is being made externally, not from client to a server. I am experimenting with a Raspberry Pi display, the app runs on Pi and listens for POST requests from my PC. When this request arrives, the image on display should refresh. So if I do it in pure Sveltekit, a request arrives to a server endpoint and then it probably should modify some shared with a client store, but since session stores are removed I don't see a way how to do it without hacks or Tauri ad-hocs.

**Reply by david-plugge** [reply]
Very interesting use case!

I agree, the tauri docs are very limited. I'm excited for v2, the docs look a lot cleaner already.
